File: sorting/insertionsort/python/insertionSort.py

def Sort(input : list) -> list:
    if len(input)<=1:
        return input

    for i in range(1,len(input)):
        for j in range(i-1,-1,-1):
            #print(j)
            if input[j+1]>input[j]:
                break
            temp = input[j+1]
            input[j+1] = input[j]
            input[j] = temp
            #print(input)

    return input

File: sorting/insertionsort/javascript/insertionSort.js

function Sort(input)
{
  if(input.length<=1)
  {
    return input;
  }
  for(let i=1;i<input.length;i++)
  {
    for(let j=i-1;j>=0;j--)
    {
      if(input[j+1]>input[j])
      {
        break;
      }
      let temp = input[j+1];
      input[j+1] = input[j];
      input[j] = temp;
    }
  }
  return input;
}

module.exports = {Sort};
